app/controllers/api/v1/send_sms_controller.rb

```diff
@@ -1,62 +1,41 @@
-require './app/services/immunization_service/send_sms_service'
 require "uri"
 require "json"
 require "net/http"
 
 class Api::V1::SendSmsController < ApplicationController
-  before_action :initialize_variables, only: [:index]
+  before_action :initialize_variables, only: [:index, :fetch_phone]
 
   def index
+    return render json: { message: "SMS reminder turned off" } if sms_reminder_off?
 
-    config_file = Rails.root.join('config', 'application.yml')
-    config = YAML.load_file(config_file)
-    sms_reminder = config.dig('development', 'sms_reminder')
-
-    if sms_reminder == 'false'
-      return render json: { message: "SMS reminder turned off" }
-    end
-
-    patient_details = patients_phone
-    result = validatephone(patient_details[:cell_phone])
-    if result.length == 13
-      patient_details[:cell_phone] = result
-      output = enqueue_sms(params[:appointment_date], patient_details)
-    else
-      output = result
-    end
+    output = process_sms_request
     render json: { message: output }
   end
 
-  def show
+  def fetch_phone
+    phone_number = validated_phone_number
+    render json: { message: phone_number }
+  end
 
+  def show
     config_file = Rails.root.join('config', 'application.yml')
-         config = YAML.load_file(config_file)       
+    config = YAML.load_file(config_file)       
       
-      begin
-
-         if config.key?('development')
-          render json: config['development']
-        else
-          render json: { error: 'Development configuration not found' }, status: :not_found
-        end
-
-      rescue Errno::ENOENT
-        render json: { error: 'Configuration file not found' }, status: :not_found
-      end
-
+    if config.key?('development')
+      render json: config['development']
+    else
+      render json: { error: 'Development configuration not found' }, status: :not_found
+    end
+  rescue Errno::ENOENT
+    render json: { error: 'Configuration file not found' }, status: :not_found
   end
 
   def update
-   
     config_file = Rails.root.join('config', 'application.yml')
     config = YAML.load_file(config_file)
 
     params.each do |key, value|
-
-      if config['development'].key?(key)
-         config['development'][key] = value
-      end
-      
+      config['development'][key] = value if config['development'].key?(key)
```

Comment:
Let us not tied it to a specific environment

Reply:
I have fixed this what i have done it checks the environment being used and then it works with that

Reply:
I have added the fix which was overwriting application.yml file contents

Reply:
I have removed the unnecessary code for checking rails enviroment